src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -369,7 +395,69 @@ public bool IsPackageDeprecated
             }
         }
 
-        public Uri IconUrl { get; set; }
+        private Uri _iconUrl;
+        public Uri IconUrl
+        {
+            get { return _iconUrl; }
+            set
+            {
+                _iconUrl = value;
+                OnPropertyChanged(nameof(IconUrl));
+            }
+        }
+
+        private IconBitmapStatus _bitmapStatus;
+
+        public IconBitmapStatus BitmapStatus
+        {
+            get { return _bitmapStatus; }
+            set
+            {
+                _bitmapStatus = value;
+                OnPropertyChanged(nameof(BitmapStatus));
+            }
+        }
+
+
+        private BitmapSource _iconBitmap;
+        public BitmapSource IconBitmap
+        {
+            get
+            {
+                if (_iconBitmap == null)
+                {
+                    if (BitmapStatus == IconBitmapStatus.None)
+                    {
+                        BitmapStatus = IconBitmapStatus.InProgress;
+                        BitmapSource iconBitmap = GetBitmapSourceOrSetStatusAsNeedToFetch();
+
+                        if (iconBitmap != null)
+                        {
+                            IconBitmap = iconBitmap;
+                        }
+                        else
+                        {
+                            NuGetUIThreadHelper.JoinableTaskFactory
+                                .RunAsync(FetchThenSetIconBitmapAsync)
+                                .PostOnFailure(nameof(PackageItemListViewModel), nameof(IconBitmap));
+
+                            // use defaultIcon until the FetchFinishes.
+                            IconBitmap = Images.DefaultPackageIcon;
```

Comment:
Should this be the before the JTF call, and also should we actually set the Status to a fetching state as well?

Reply:
not positive on the order. my thought was to kick off the work fast...either way the propertychange handling won't happen until after both are done anyway, i beliee.

i once had a fetching state...but it is not needed and adds some complexity. could add back or rename needtofetch if u think it is critical.

Reply:
ok.. i am switching the order. had thought that the background work would always lose to line 445...but not always the case.

Reply:
fixed in current iteration.